runtime/lib/common/system_desc.cpp

```diff
@@ -29,6 +29,7 @@
 #pragma clang diagnostic ignored "-Wunused-function"
 #pragma clang diagnostic ignored "-Wunused-local-typedef"
 #pragma clang diagnostic ignored "-Wzero-length-array"
+#pragma clang diagnostic ignored "-Wshift-op-parentheses"
```

Comment:
😭

Reply:
Yeah. Coming from tt-metal like previous cases we encountered.  Should we:

1. Ignore for now (when not tt-mlir issue), follow up on all of these at some later date (someone could try removing, and drive closure on issues)
2. File new metal issues for every one encountered (are they expecting/able to support it?)
3. File single metal issue and keep piling things into it for cases like this

(Leaning toward 1 for now, to keep this uplift process light weight until we have "dedicated owner")

Reply:
I think yeah 1 or 3.  Maybe we can sync with someone from metal.

@blozano-tt / @afuller-TT / @ayerofieiev-tt, do you guys want us to file issues for these?  I think we have enabled a different set of warnings.  So far we've just been ignoring warnings that come up.

Reply:
Surface-level:
* If it's important/urgent, then create an issue and poke us/someone so we know it's urgent. 
* If it's not important for now, happy to ignore -- we are doing some cleanup (just got clang-tidy scan landed and have a long list of TO-DO's there), so we are making forward progress, but in no particular order (`performance` and `bugprone` are top of list, though).  The [list of disabled warnings](https://github.com/tenstorrent/tt-metal/blob/main/CMakeLists.txt#L222-L242) are another TODO list with intention but no timeline. 
* If filing tickets, please a single ticket with a list of which are important would be easier to manage than juggling a dozen.

Digging deeper:
All this pragma push gymnastics is only because the header violates these warnings, right?  Pretty sure clang should be far more lenient with diagnostics in system headers.  How are you consuming `tt-metal` right now?  Is the include path not flagged as `SYSTEM` in CMake?  If not, I'm happy to help there.  That should insulate you from issues in the headers.

Reply:
> If filing tickets, please a single ticket with a list of which are important would be easier to manage than juggling a dozen.

Sounds good.

> How are you consuming tt-metal right now? Is the include path not flagged as SYSTEM in CMake?
We are not using `SYSTEM` or anything fancy.  We are just doing an `ExternalProject_Add`, it'd be great to get your input if there's a cleaner way for us to consume metal! See our cmake file for metal:
https://github.com/tenstorrent/tt-mlir/blob/main/third_party/CMakeLists.txt

Reply:
I took a quick peek.  First of all: sorry you have to jump through these hoops for tt-metal.  It's on the roadmap to have it properly consumable.  But that's still a ways out.

In the meantime, what you have looks pretty good with one significant suggestion:
Include dirs [passed as a variable](https://github.com/tenstorrent/tt-mlir/blob/main/third_party/CMakeLists.txt#L19-L37) to be picked up who-knows-where is frustrating to reason about.  But it also hinders flagging their origin as "not our code".  Which is what's biting you here.
Instead, attach them to your `tt-metal` target and include the `SYSTEM` flag, like so:

```cmake
target_include_directories( tt-metal SYSTEM
  INTERFACE
    foo
    bar
    baz
)
```

Consumers should depend on `tt-metal` via a `target_link_libraries` (which unfortunate for the naming does far more than just link -- it'll handle all the transitive usage requirements like the include dirs).  Any targets that were magically getting the include dirs from the variable and _not_ having a dependency on the imported target will need to have that corrected.  If there are any.

With that in place if you inspect the compile line you should start to see `-isystem` for these paths.  Then try removing these diagnostics overrides and it should come up clean.  98 ¾ percent guaranteed.

I don't have a tt-mlir build setup on my machine right now, but I can give that a whirl next week if you don't get to it first.

Reply:
Thank you @afuller-TT, this did the trick for me locally! Testing it in CI: https://github.com/tenstorrent/tt-mlir/pull/1309

Reply:
Nick's PR merged, marking this resolved.  Thanks a lot, Andrew for feedback!